kcalcore/recurrencerule.cpp

```diff
@@ -623,6 +623,16 @@ void Constraint::appendDateTime( const QDate &date, const QTime &time,
                                  QList<KDateTime> &list ) const
 {
   KDateTime dt( date, time, timespec );
+  KDateTime nonTimespecDt( date, time, KDateTime::Spec::UTC() );
+  if ( !dt.isValid() && nonTimespecDt.isValid() && timespec.utcOffset() == 0 ) {
+    // This is to fix an issue where CalDAV servers can provide a VTIMEZONE
+    // definition for CEST (eg "Europe/Berlin") with a DAYLIGHT component like:
+    // "DTSTART:19700329T020000\r\n"
+    // "RRULE:FREQ=YEARLY;BYDAY=-1SU;BYMONTH=3\r\n"
+    // This will fail after 1981 since the system definition differs, resulting
+    // in an invalid KDateTime due to the "invalid" timespec.
```

Comment:
`timespec.utcOffset()` on has useful information if the `SpecType` is `OffsetFromUTC`, which is probably not the case, cf. https://github.com/mer-packages/kcalcore/blob/master/kcalcore/kdedate/kdatetime.h#L322

But then again, we need to have UTC times, so we do not run into invalid times due to skipped hours when switching to DST. I have created an according pull request that tries to fix the problem one level above. Please let me know what you think about it. Thanks!